Sources/FBLPromises/include/FBLPromise+Retry.h

```diff
@@ -0,0 +1,160 @@
+/**
+ Copyright 2018 Google Inc. All rights reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at:
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "FBLPromise.h"
```

Comment:
May want to mention somewhere that having retry `count` <= 0 will still perform the work once. Otherwise it can be confusing that something like `+[FBLPromise attempts:0 retry:work]` still does something.

I wonder if the method naming would be clearer as `+[FBLPromise do:work retryAttempts:0]` and `+[FBLPromise do:work retryAttempts:0 afterDelay:delay]`?

Reply:
Updated documentation.

Definitely see your point, since `retry` allows specifying quite a few parameters, it does make the fully overloaded ObjC method a bit more difficult to read, i.e.
[FBLPromise onQueue:attempts:delay:condition:retry:], but the general convention is to have the `work` block last to make it easier to to inline the block code and utilize trailing closure syntax in Swift.

Reply:
Even with the work block last, it seems that naming the work param `retry` can be misleading since it executes even if not retried. Would something like `+[FBLPromise onQueue:retryAttempts:delay:condition:work:]` be clearer?

Reply:
What about the naming for:
`+[FBLPromise retry:]` and `+[FBLPromise onQueue:retry:]`
where the default retry attempts count and delay interval are used?